Dbpedia History (#745)
dbpedia/extraction-framework: history/ReadMe.md

# DBPEDIA HISTORY

> DBpedia History allows to extract into a RDF format the history of Wikipedia chapter

Reviewer: ```suggestion
DBpedia History enables the history of a Wikipedia chapter to be extracted into an RDF format 
```

## Previous work

> This DBpedia App is a scala/java version of the first work conducted by the French Chapter : https://github.com/dbpedia/Historic/ 

Reviewer: ```suggestion
This DBpedia App is a Scala/Java version of the first work conducted by the French Chapter, <https://github.com/dbpedia/Historic/>.
```

Fabien Gandon, Raphael Boyer, Olivier Corby, Alexandre Monnin. Wikipedia editing history in DBpedia : extracting and publishing the encyclopedia editing activity as linked data. IEEE/WIC/ACM International Joint Conference on Web Intelligence (WI' 16), Oct 2016, Omaha, United States. <hal-01359575>
https://hal.inria.fr/hal-01359575

Fabien Gandon, Raphael Boyer, Olivier Corby, Alexandre Monnin. Materializing the editing history of Wikipedia as linked data in DBpedia. ISWC 2016 - 15th International Semantic Web Conference, Oct 2016, Kobe, Japan. <http://iswc2016.semanticweb.org/>. <hal-01359583>
https://hal.inria.fr/hal-01359583

## A first working prototype

> This prototype is not optimized, during its development of it we were faced with the WikiPage type checking constraints that are checked in almost every module of the DBpedia pipeline.
> We hardly copy/paste and renamed all the classes and objects we needed for running the extractors.
> This conception could be easily improved by making WikiPage and WikiPageWithRevision objects inherit from the same abstract object.

Reviewer: I guessed at what was intended by the adverb `hardly` in the second line. 
```suggestion
This prototype is not optimized. During its development, we were faced with the WikiPage type-checking constraints that are checked in almost every module of the DBpedia pipeline.
We basically copy/pasted and renamed all the classes and objects we needed for running the extractors.
This conception could be easily improved by making `WikiPage` and `WikiPageWithRevision` objects inherit from the same abstract object.
```

> But as a first step, we wanted to touch the less possible DBpedia core module.

Reviewer: I cannot tell what was intended by `touch the less possible`... This needs a better translation.

Author: I changed it by writing "we didn't want to impact the core module", is it better ?

> Some other improvements that could be conducted:

Reviewer: ```suggestion
Some other improvements that could be made:
```

* Scala version

> * Being able to use a historic namespace taking into account the DBpedia chapter language
> * Being able to follow if a revision impacts an infobox content

Reviewer: ```suggestion
* Enabling use of a historic namespace, taking into account the DBpedia chapter language
* Enabling following when a revision impacts content of an `infobox`
```

## Main Class

* [WikipediaDumpParserHistory.java](src/main/java/org/dbpedia/extraction/sources/WikipediaDumpParserHistory.java) : for parsing of the history dumps
* [RevisionNode.scala](src/main/scala/org/dbpedia/extraction/wikiparser/RevisionNode.scala)  : define revision node object
* [WikiPageWithRevision](src/main/scala/org/dbpedia/extraction/wikiparser/WikiPageWithRevisions.scala) : define wikipage with revision list object 

## Extractors 

> * [HistoryPageExtractor.scala](src/main/scala/org/dbpedia/extraction/mappings/HistoryPageExtractor.scala): Extract all the revision of every wikipedia pages
> * [HistoryStatsExtractor.scala](src/main/scala/org/dbpedia/extraction/mappings/HistoryStatsExtractor.scala) : Extract statistics about the revision activity for every page of Wikipedia

Reviewer: ```suggestion
* [HistoryPageExtractor.scala](src/main/scala/org/dbpedia/extraction/mappings/HistoryPageExtractor.scala) — Extract all revisions of every Wikipedia page
* [HistoryStatsExtractor.scala](src/main/scala/org/dbpedia/extraction/mappings/HistoryStatsExtractor.scala) — Extract statistics about revision activity for every page of Wikipedia
```

* configure the [extraction.properties](extraction.properties) file
* and run  ```../run run extraction.properties```

> * Test it with `mvn test` (need to have a containing file, `frwiki-[YYYYMMDD]-download-complete` empty flag file into the `base-dir` defined into the `extraction-properties` file)

Reviewer: This could use some rewording, as it's not clear what's meant by what's here. (I made some tweaks that may _help_ with clarity, but they're insufficient on their own.)

* configure the [extraction.properties](extraction.properties) file
* and run  ```../run run extraction.properties```

> * Test it with `mvn test`, if you cold start you may have to artificially create a `frwiki-[YYYYMMDD]-download-complete` empty file into the `base-dir` defined into the `extraction-properties` file

Reviewer: ```suggestion
* Test it with `mvn test`. If you're starting cold, you may have to manually create an empty file named in the form of `frwiki-[YYYYMMDD]-download-complete` (for instance, `frwiki-20221209-download-complete`) in the `base-dir` as defined in the `extraction-properties` file.
```

* and run  ```../run run extraction.properties```

* Test it with `mvn test`, if you cold start you may have to artificially create a `frwiki-[YYYYMMDD]-download-complete` empty file into the `base-dir` defined into the `extraction-properties` file

### Triple extracted

> Given this little wikipedia page : [Hôtes_de_passage](https://fr.wikipedia.org/wiki/H%C3%B4tes_de_passage)
> 
> -> The HistoryPageExtractor.scala extractor will produce : 

Reviewer: ```suggestion
Given this little Wikipedia page: [Hôtes_de_passage](https://fr.wikipedia.org/wiki/H%C3%B4tes_de_passage)

→ The `HistoryPageExtractor.scala` extractor will produce: 
```

<http://fr.wikipedia.org/wiki/Hôtes_de_passage?oldid=36815850&ns=0> <http://purl.org/dc/terms/creator> <http://fr.dbpedia.org/resource/Hôtes_de_passage__creator__82.244.44.195> .
<http://fr.dbpedia.org/resource/Hôtes_de_passage__creator__82.244.44.195> <http://rdfs.org/sioc/ns#ip_address> "82.244.44.195" .
<http://fr.wikipedia.org/wiki/Hôtes_de_passage?oldid=36815850&ns=0> <http://rdfs.org/sioc/ns#note> "Nouvelle page : ''Hôtes de passage'' appartient est une partie de ''La Corde et les souris'' consignée dans ''Le Miroir des limbes''.Malraux y mène des entretiens avec entre autre Senghor et un p..." .
<http://fr.wikipedia.org/wiki/Hôtes_de_passage?oldid=36815850&ns=0> <http://dbpedia.org/ontology/wikiPageLength> "214"^^<http://www.w3.org/2001/XMLSchema#nonNegativeInteger> .
<http://fr.wikipedia.org/wiki/Hôtes_de_passage?oldid=36815850&ns=0> <http://dbpedia.org/ontology/wikiPageLengthDelta> "214"^^<http://www.w3.org/2001/XMLSchema#integer> .
<http://fr.wikipedia.org/wiki/Hôtes_de_passage?oldid=36815850&ns=0> <http://dbpedia.org/ontology/isMinorRevision> "false"^^<http://www.w3.org/2001/XMLSchema#boolean> .
.....
```

> -> HistoryStatsExtractor.scala extractor will produce : 

Reviewer: ~~~suggestion
...
```

→ `HistoryStatsExtractor.scala` extractor will produce: 
~~~
